src/modules/user-auth/components/RegisterForm/useRegisterForm.ts

```diff
@@ -13,47 +13,19 @@ import { useForm } from 'react-hook-form';
 
 export default function useRegisterForm() {
   const { t } = useTranslation();
+  const { registerModal } = useLayoutModal();
+  const [submitError, setSubmitError] = useState('');
 
   const formSchema = RegisterFormSchema(t);
   const form = useForm<RegisterFormModel>({
     defaultValues: formSchema.getDefault(),
     resolver: yupResolver(formSchema),
   });
-  const { control } = form;
-
-  const [submitError, setSubmitError] = useState('');
-
-  const controllers: Controllers<RegisterFormModel> = {
-    firstName: {
-      control,
-      name: 'firstName',
-    },
-    lastName: {
-      control,
-      name: 'lastName',
-    },
-    gender: {
-      control,
-      name: 'gender',
-    },
-    dob: {
-      control,
-      name: 'dob',
-    },
-    email: {
-      control,
-      name: 'email',
-    },
-    password: {
-      control,
-      name: 'password',
-    },
-  };
 
   const onSubmit = form.handleSubmit(async (data) => {
-    await AuthService.registerWithEmail(data).catch((e) =>
-      setSubmitError(getErrorMessage(e, t))
-    );
+    AuthService.registerWithEmail(data)
+      .then(() => registerModal.hide())
```

Comment:
use `await`

Reply:
Same as above, don't think we should do this